src/twisted/web/http_headers.py

```diff
@@ -235,12 +267,14 @@ def addRawHeader(self, name, value):
 
         self.setRawHeaders(name, values)
 
-    def getRawHeaders(self, name, default=None):
+    def getRawHeaders(
+        self, name: String, default: Optional[Any] = None
+    ) -> Optional[Union[List[String], Any]]:
```

Comment:
[`typing.overload`](https://docs.python.org/3/library/typing.html#typing.overload) could be used to express that the returned string type matches `name`. If this were used in combination with a `TypeVar` for the type of `default`, the return type would be exact.

Reply:
@wsanchez @twm What is the best way to use `typing.overload` here?  Will this still retain compatibility with things like klein and treq?